src/app/util/bytes-data.h

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+
+/**
+ * @brief A warpper class for holding bytes buffer and its length, without the ownershap of buffer.
+ */
+class BytesData
```

Comment:
A few things:

1) If this name is meant to be data-model-specific, why not just call it OctetString?
2) If this name is not meant to be data-model-specific, `template <typename T> Span` is probably more appropriate.  We could still typedef something like `OctetString` or `ByteData` (note singular) of the `uint8_t` case if we don't want to write `Span<uint8_t>` everywhere.

Reply:
Moved this to src/lib/support, according to the current and future usage, maybe I prefer not to use a template since the CHIPTLV does not really cares about the length of data when encoding it.